posthog/temporal/data_imports/sources/linkedin_ads/linkedin_ads.py

```diff
@@ -0,0 +1,317 @@
+import json
+import typing
+import datetime as dt
+import collections.abc
+
+import pyarrow as pa
+from dlt.common.normalizers.naming.snake_case import NamingConvention
+
+from posthog.models.integration import Integration
+from posthog.temporal.data_imports.pipelines.helpers import incremental_type_to_initial_value
+from posthog.temporal.data_imports.pipelines.pipeline.typings import SourceResponse
+from posthog.temporal.data_imports.sources.common.sql import Column, Table
+from posthog.temporal.data_imports.sources.generated_configs import LinkedinAdsSourceConfig
+from posthog.warehouse.types import IncrementalFieldType
+
+from .client import LINKEDIN_SPONSORED_URN_PREFIX, LinkedinAdsClient, LinkedinAdsResource
+from .schemas import (
+    DATE_FIELDS,
+    FLOAT_FIELDS,
+    INTEGER_FIELDS,
+    RESOURCE_SCHEMAS,
+    RESOURCE_VIRTUAL_COLUMNS,
+    VIRTUAL_COLUMN_URN_MAPPING,
+    VIRTUAL_COLUMNS,
+)
+
+
+class LinkedinAdsColumn(Column):
+    """Represents a column of a LinkedIn Ads resource."""
+
+    def __init__(self, qualified_name: str, data_type: str = "string"):
+        self.name = qualified_name.replace(".", "_")
+        self.qualified_name = qualified_name
+        self.data_type = data_type
+
+    def to_arrow_field(self):
+        """Return the Arrow type associated with this column."""
+        arrow_type: pa.DataType
+
+        if self.qualified_name in INTEGER_FIELDS:
+            arrow_type = pa.int64()
+        elif self.qualified_name in FLOAT_FIELDS:
+            arrow_type = pa.float64()
+        elif self.qualified_name in DATE_FIELDS:
+            arrow_type = pa.date32()
+        else:
+            arrow_type = pa.string()
+
+        return pa.field(self.name, arrow_type)
```

Comment:
You don't need all of this btw, because LinkedIn is a API response, the results can vary, and so setting a strict means we have to keep it updated. Of course, you can do this if you need a strict schema in web analytics land, but one to think about